### PR TITLE
Implement SCRAM auth

### PR DESCRIPTION
**Pull request by EinKrebs**

### CODE REVIEW

File: pkg/auth/auth.go [hunk 1]
```diff
@@ -53,6 +54,61 @@ func AuthBackend(shard conn.DBInstance, berule *config.BackendRule, msg pgproto3
 			return fmt.Errorf("no auth rule specified for server connection")
 		}
 		return shard.Send(&pgproto3.PasswordMessage{Password: berule.AuthRule.Password})
```
Comment (reshke) [inline pkg/auth/auth.go:56]: what about authFrontend() ? 

also, we can actually setup postgresql shard with scram auth in tests and run scram auth in CI....

**Comment by reshke** [comment]
Also pleasse indicate in the documentation/readme that SCRAM is now supported 🦀️️️️️️🦀️️️️️️🦀️️️️️️

**Comment by EinKrebs** [comment]
> Also pleasse indicate in the documentation/readme that SCRAM is now supported crab️️️️️crab️️️️️crab️️️️️

Where to write this? Do I need to create another .md file in docs/ or update existing?

**Comment by Denchick** [comment]
> Where to write this? Do I need to create another .md file in docs/ or update existing?

I would suggest create a new section Authentication methods or something in Router.md file.

**Comment by reshke** [comment]
So, basically scram test are the same as non-scram test with exception in auth process be each sql-file run? Maybe we can add unit test for this feature too? See https://github.com/pg-sharding/spqr/blob/master/test/authentification/auth_test.go

**Comment by EinKrebs** [comment]
> So, basically scram test are the same as non-scram test with exception in auth process be each sql-file run? Maybe we can add unit test for this feature too? See https://github.com/pg-sharding/spqr/blob/master/test/authentification/auth_test.go

It seems rather hard to mock psql's behavior in unit tests, it's way easier to set up cluster in docker compose

**Review by reshke — APPROVED** [review]

**Review by Denchick — CHANGES_REQUESTED** [review]
Let's not test SCRUM in docker

**Review by Denchick — APPROVED** [review]

**Pull request merged**